pkg/storage/db.go

```diff
@@ -293,23 +268,66 @@ func (db *DB) runQuery(ctx context.Context, query string, args ...interface{}) (
 
 	defer rows.Close()
 
-	networkChangeEvents := make([]domain.NetworkChangeEvent, 0)
+	cloudAssetDetails := make([]domain.CloudAssetDetails, 0)
+
+	tempMap := make(map[string]*domain.CloudAssetDetails)
```

Comment:
Why a map of `*domain.CloudAssetDetails` rather than the value-type version?

Reply:
Mostly so I could do [this nil check and just assign the address](https://github.com/asecurityteam/asset-inventory-api/blob/feature/api-to-db-integration/pkg/storage/db.go#L293-L294) of the newly created `row` struct.

If I change it by making it a value rather than a pointer, like:
```
tempMap := make(map[string]domain.CloudAssetDetails)
```

Then I have to change the if clause linked above like so:
```
if _, existingValue := tempMap[row.ResourceID]; !existingValue {
	tempMap[row.ResourceID] = row
}
```

And I got more code to change later when the map is no longer a map of pointers, and [this SO question and answer explains](https://stackoverflow.com/questions/32751537/why-do-i-get-a-cannot-assign-error-when-setting-value-to-a-struct-as-a-value-i) it:

![image](https://user-images.githubusercontent.com/857712/56701958-3d087400-66c7-11e9-8923-279914bfe894.png)

My uneducated preference would be to keep this code as-is, but if there really is a pitfall I'm gonna hit, of course I'll change it.

Reply:
Ah okay. that makes sense 👍